core/io/darwin.rs

```diff
@@ -51,6 +51,89 @@ impl IO for DarwinIO {
         Ok(darwin_file)
     }
 
+    fn wait_for_completion(&self, timeout: i32) -> Result<()> {
```

Comment:
If this method added to the `IO` trait stays, I will need someone to test this on Mac.